### DISCUSSION TITLE
Filter processor

### DISCUSSION THREAD

**Discussion by psd2ibanity** [open-telemetry/opentelemetry-erlang#554] [Q&A] [answered]
Hi,

We have successfully integrated OpenTelemetry libs into our Elixir code. As the default configuration states, we have only one processors for traces, which is :batch (otel_batch_processor). I was wondering if it's either already possible, or will be possible one day, to use other processors, especially filter/transform processor (https://opentelemetry.io/docs/collector/transforming-telemetry/) in order to clean sensitive attributes from our traces?
Thanks

**Comment by tsloughter** [comment, marked as answer]
Sadly processors in the collector (what you linked to) and in the SDK are different, but similar.

While you can today define multiple processors in your Elixir config you can't transform/filter the spans `OnEnd`, by spec design. In https://github.com/open-telemetry/opentelemetry-erlang-contrib/blob/bc11851360ce7b4ca73de7ea544b503366e7e118/examples/basic_elixir/config/releases.exs you can see how `processors` is configured as a list, but the returned Span in one processors `OnEnd` is no passed to `OnEnd` of the next processor in the list, so filtering/transforming would be lost.

There has been off and on discussion on how to add such functionality to the Spec, see

https://github.com/open-telemetry/opentelemetry-specification/pull/3189

and 

https://github.com/open-telemetry/opentelemetry-specification/issues/1089

**Comment by psd2ibanity** [comment]
Thank you for the complete answer.
